FirebaseMessaging/Sources/FIRMessagingAnalytics.m

```diff
@@ -212,11 +216,34 @@ + (void)logMessage:(NSDictionary *)notification
       break;
 
     default:
-      // Only a silent notification (i.e. 'content-available' is true) can be received while the app
-      // is in the background. These messages aren't loggable anyway.
+      // App was either in background state or in transition from closed
+      // to open.
+      // Needs to check notification is displayed.
+      if ([[self class] isDisplayNotification:notification]) {
+        [self logOpenNotification:notification toAnalytics:analytics];
+      }
       break;
   }
 #endif
 }
 
++ (BOOL)isDisplayNotification:(NSDictionary *)notification {
+  NSDictionary *aps = notification[kApsKey];
+  if (!aps || ![aps isKindOfClass:[NSDictionary class]]) {
+    return NO;
+  }
+  NSDictionary *alert = aps[kApsAlertKey];
+  if (!alert) {
+    return NO;
+  }
+  if ([alert isKindOfClass:[NSDictionary class]]) {
```

Comment:
How is it determined that the next two cases are the only ones that we want to log analytics for?

Reply:
We want to differentiate display notifications from hidden ones, so the field "alert" must exist and have at least one value in it.